### PR TITLE
Feature/service ratelimit

### PR DESCRIPTION
**Pull request by szuecs**
Rate limit type service implements #470 a backend protection. 

It allows the definition of a maximum amount of requests that are passed to the backend without considering data from the client. It would be used like this:

    ratelimit(200, "1s")

The naming ratelimit() and not serviceRatelimit() indicates, that this is likely the default behaviour a user wants to have. Instead of the more complex localRatelimit(), which allows rate limiting based on client data.

### CODE REVIEW

File: filters/ratelimit/ratelimit.go [hunk 1]
```diff
@@ -46,6 +46,19 @@ func NewLocalRatelimit() filters.Spec {
...
+//
+// Example:
+//
+//    backendHealthcheck: Path("/healthcheck")
+//    -> ratelimit(20, "1s")
+//    -> "https://foo.backend.net";
+func NewRatelimit() filters.Spec {
+	return &spec{typ: ratelimit.ServiceRatelimit, filterName: ratelimit.ServiceRatelimitName}
+}
+
```
Comment (ruiaraujo) [inline filters/ratelimit/ratelimit.go]: ❤️

File: ratelimit/ratelimit.go [hunk 2]
```diff
@@ -25,8 +27,13 @@ const (
 const (
 	// NoRatelimit is not used
 	NoRatelimit Type = iota
-	// LocalRatelimit is used to have a simple local rate limit,
-	// which is calculated and measured within each instance
+	// ServiceRatelimit is used to have a simple rate limit for a
```
Comment (aryszka) [inline ratelimit/ratelimit.go]: i would rephrase the scope of this rate limiter somehow. As i understand, it is a single global bucket, and it can be used in a backend specific way, if defined for a particular route.

**Comment by aryszka** [comment]
:+1:

**Comment by ruiaraujo** [comment]
:+1:
